### ISSUE TITLE
ghc-8.6: makePackages.sh dies at wrong patch for filepath

### ISSUE THREAD

**Issue by Mikolaj** [ghcjs/ghcjs#720]
```
./utils/makePackages.sh
[...]
Warning: Cannot run preprocessors. Run 'configure' command first.
Building source dist for filepath-1.4.2.1...
Source tarball created: dist-install/filepath-1.4.2.1.tar.gz
unpacking sources: /home/mikolaj/r/ghcjs/utils/../ghc/libraries/filepath/dist-install/filepath-1.4.2.1.tar.gz
patching package: filepath
patching file filepath.cabal
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
3 out of 3 hunks ignored -- saving rejects to file filepath.cabal.rej
patching file prologue.txt
patching file README.md
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file README.md.rej
patching file System/FilePath/Current.hs
patching file System/FilePath/Internal.hs
patching file travis.hs
```

With ghc-8.4 it applies fine.

**Comment by luite** [comment]
hmm, this may happen in a tree where you've already run the `makePackages.sh` script.

If this also happens in a clean tree then I must have messed up the patch.

**Comment by Mikolaj** [comment]
I've just checked again, with the same result:

```
rm -rf ghcjs
git clone --branch ghc-8.6 https://github.com/ghcjs/ghcjs.git
cd ghcjs
git submodule update --init --recursive
./utils/makePackages.sh
```

**Comment by Mikolaj** [comment]
BTW, GHC is 8.6.3, if that matters.

**Comment by luite** [comment]
hmm, I see the problem too, and it's not a patch that I forgot to update or commit. running `utils/updatePatches.sh` gives me the same patch again, from a working tree that I just ran the testsuite in.

I'll investigate after some sleep, since it's rather late here.

**Comment by Mikolaj** [comment]
Thank you. No hurry. :)

**Comment by luite** [comment]
Oh I see what I did wrong! After updating the `ghc` submodule, I didn't regenerate the `lib/upstream` directory (populated by `makePackages.sh`).

**Comment by sheyll** [comment]
totally of topic: _(:  at least that was probably about 6h of sleep_, anyway **thanks** for your persistent (and hopefully sustainable) contribution!
I just love to see that GHCJS is alive, since we decided to use it in our next project, to replace a big piece of flex and java/wicket code.

**Comment by luite** [comment]
thanks, and fixed by e422fbb4acecd124f118a49119d2577eb1536fa0

**Comment by Mikolaj** [comment]
Thank you. I confirm ghcjs with 8.6 branch compiles now for me and moreover LambdaHack runs with it without any crashes (so far).
